agents/holo_synth/synth_agent.py

```diff
@@ -0,0 +1,208 @@
+            # Locking mechanism stops code from proceeding if no lock acquired
+            if not acquired:
+                self.log.warn(
+                    "Could not start init because {} is already running".format(
+                        self.lock.job
+                    )
+                )
+                return False, "Could not acquire lock."
+            # Run the function you want to run
+            self.log.debug("Lock Acquired Connecting to Stages")
+            self.lo_id = synth3.get_LOs()
+
+            synth3.set_RF_output(0, 1, self.lo_id) # LO ID, On=1, USB connection ID
+            synth3.set_RF_output(1, 1, self.lo_id) # LO ID, On=1, USB connection ID
+
+            # data = {"timestamp": time.time(), "block_name": "synth_LO", "data": {}}
+
+            # data["data"]["F1_status"] = 1
+            # data["data"]["F2_status"] = 1
+
+            # self.agent.publish_to_feed("synth_LO", data)
+            # session.data.update(data["data"])
+
+
+        # This part is for the record and to allow future calls to proceed,
+        # so does not require the lock
+        self.initialized = True
+
+        return True, "Synth Initialized."
+
+    def set_frequencies(self, session, params):
+        """
+        params: 
+            dict: {'freq0': float
+                   'freq1': float}
+        """
+
+        f1 = params.get("freq1", 0)
+        f_offset = params.get("offset", 0)
+
+        F_1 = int(f1 * self.ghz_to_mhz / self.N_MULT) # Convert GHz -> MHz for synthesizers
+
+        print(F_1,F_1+f_offset)
+
+        with self.lock.acquire_timeout(timeout=3, job="set_frqeuencies") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+            synth3.set_f(0, F_1, self.lo_id)
+            synth3.set_f(1, F_1+f_offset, self.lo_id)
+
+            data = {"timestamp": time.time(), "block_name": "synth_LO", "data": {}}
+
+            data["data"]["F1"] = F_1
+            data["data"]["F2"] = F_1+f_offset
+
+            self.agent.publish_to_feed("synth_LO", data)
+            session.data.update(data["data"])
+
+        return True, "Frequencies Updated"
+
+    # This function is not finished, need to figure out how to read out frequency from USB connection.
+    def read_frequencies(self, session, params=None):
+
+        with self.lock.acquire_timeout(timeout=3, job="read_frqeuencies") as acquired:
+            if not acquired:
+                self.log.warn(
+                    f"Could not set position because lock held by {self.lock.job}"
+                )
+                return False, "Could not acquire lock"
+
+        return True, "Frequencies Updated"
+
+    def set_synth_status(self, session, params):
+        """
+        params: 
+            dict: {'LO_ID': int
+                   'status': int}
```

Comment:
'LO_ID' is used in this docstring, while "lo_id" is used in the code. Make consistent.